retrievalmarket/types.go

```diff
@@ -260,15 +249,71 @@ func (p Params) IntervalLowerBound(currentInterval uint64) uint64 {
 	intervalSize := p.PaymentInterval
 	var lowerBound uint64
 	var target uint64
-	for target < currentInterval {
+	for target <= currentInterval {
 		lowerBound = target
 		target += intervalSize
 		intervalSize += p.PaymentIntervalIncrease
 	}
 	return lowerBound
 }
 
-func (p Params) NextInterval(currentInterval uint64) uint64 {
+// OutstandingBalance produces the amount owed based on the deal params
+// for the given transfer state and funds received
```

Comment:
"given transfer state" - so I see that this method returns the oustanding unsealed price as the "outstanding balance" but if the unsealing price is covered then the "outstanding balance" is something different. So this is really an "OutstandingBalanceForState", is that correct? (no need to rename, you say this in the comment, I'm just making sure I'm reading this right).

Reply:
yes. amount transferred will always be zero (or should always be zero) if the unseal price hasn't been paid, as unsealing won't have even begun, which means that transfer will not have begun. :)

I'll do some commenting here.